src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -275,13 +275,13 @@ private static JObject ReadProject(string projectJsonPath)
            } 
         }
 
-        private JToken GetFrameworkDependenciesSection(JObject projectJsonRoot, string framework = null)
+        private JObject GetFrameworkDependenciesSection(JObject projectJsonRoot, string framework = null)
         {
             if(string.IsNullOrWhiteSpace(framework))
             {
-                return projectJsonRoot["dependencies"];
+                return (JObject) projectJsonRoot["dependencies"];
             }
-            return projectJsonRoot.SelectToken("frameworks." + NewtonsoftEscapeJProperty(framework) + ".dependencies");
+            return (JObject) projectJsonRoot["frameworks"][framework]["dependencies"];
```

Comment:
I don't see anything about the other two instances which necessitates they be treated as JProperties.